python/py-isce2/Portfile

```diff
@@ -0,0 +1,154 @@
+
+2) One can use isce2 with other packages like SpiceyPy, sarpy etc. These will need to be installed from source separately.
+
+    "
+
+    depends_lib-append      port:curl \
+                            port:bzip2\
+                            port:openmotif\
+                            port:hdf5 \
+                            port:fftw-3\
+                            port:fftw-3-single\
+                            port:gdal \
+                            port:scons\
+                            port:py${python.version}-setuptools\
+                            port:py${python.version}-cython\
+                            port:py${python.version}-numpy \
+                            port:py${python.version}-scipy \
+                            port:py${python.version}-nose  \
+                            port:py${python.version}-gdal  \
+                            port:py${python.version}-h5py
+
+    require_active_variants port:curl ssl
+    require_active_variants port:gdal hdf5
+    require_active_variants port:gdal netcdf
+
+    post-patch {
+        system "find ${workpath} -type f -print0 | xargs -0 -L1 env LANG=C sed -i '' 's|cython3|${prefix}/bin/cython-${python.branch}|g'"
+    }
+
+    pre-configure {
+        xinstall -d ${workconfpath}
+        xinstall -d ${workbldpath}
+    }
+
+    configure {
+        #Check if compiler is set
+        if {![gcc_variant_isset]} {
+            default_variants-append +gcc7
+        }
+
+        #For gcc, set libstdc++
+        configure.cxx_stdlib libstdc++
+
+		#Setup SConfigISCE and config dir here
+        set gccvar  [gcc_variant_name]
+        set gcclast [string index ${gccvar} end]
+        set cfgname ${workconfpath}/SConfigISCE
+        set fileId [open $cfgname "w"]
+
+
+        puts   $fileId  "PRJ_SCONS_BUILD  =   ${workbldpath}"
+        puts   $fileId  "PRJ_SCONS_INSTALL =  ${workinstallpath}"
+        puts   $fileId  "LIBPATH  =      ${prefix}/lib"
+        puts   $fileId  "CPPPATH  =      ${python.include}"
+        puts   $fileId  "CC       =      ${prefix}/bin/gcc-mp-${gcclast}"
+        puts   $fileId  "CXX      =      ${prefix}/bin/g++-mp-${gcclast}"
+        puts   $fileId  "FORTRANPATH  =   ${prefix}/include"
+        puts   $fileId  "FORTRAN   =     ${prefix}/bin/gfortran-mp-${gcclast}"
+        puts   $fileId  "MOTIFLIBPATH =  ${prefix}/lib"
+        puts   $fileId  "X11LIBPATH   =  ${prefix}/lib"
+        puts   $fileId  "MOTIFINCPATH =  ${prefix}/include"
+        puts   $fileId  "X11INCPATH   =  ${prefix}/include"
+        close   $fileId
+}
+
+
+#### This will contain the linking of executables to /opt/local/bin
+#### This should also contain softlinking of python executable to /opt/local/bin/python3
+    destroot {
+        ####This is a temporary fix till "SConstruct" is fixed.
+        system "find ${workinstallpath} -name *.py -exec sed -i '' 's/usr\\\/bin\\\/env python3/usr\\\/bin\\\/env python${python.branch}/g' {} +"
```

Comment:
I again suspect we have cleaner primitives for this available.

Reply:
Same as above. Will figure out a replacement with reinplace